utils/proxy/_deserializer.py

```diff
@@ -211,16 +211,22 @@ def _deserialize_meta(span):
             meta[key] = json.loads(meta[key])
 
 
-def _convert_bytes_values(item):
+def _convert_bytes_values(item, path=""):
     if isinstance(item, dict):
         for key in item:
             if isinstance(item[key], bytes):
-                item[key] = item[key].decode("ascii")
+                try:
+                    item[key] = item[key].decode("ascii")
+                except UnicodeDecodeError as e:
+                    if path == "[][].meta_struct":
+                        item[key] = msgpack.unpackb(item[key], unicode_errors="replace", strict_map_key=False)
+                    else:
+                        raise ValueError(f"Error decoding {path}") from e
```

Comment:
@cbeauchesne I do agree this can be a problem. Reading the msgpack documentation, sending a small positive integer (it would be a mistake from the tracer, but can happen), can lead to a pure ascii message.
Would it be possible to test the path first then use ascii or msgpack depending on the path?

Reply:
it is mandatory that all values in the metastruct dictionary will be msgpack encoded. It should be the only ones too, as far as I know.